montepy/input_parser/syntax_node.py

```diff
@@ -393,7 +393,7 @@ def append(self, val, is_comment=False):
         :param is_comment: whether or not the node is a comment.
         :type is_comment: bool
         """
-        if is_comment:
+        if is_comment or isinstance(val, CommentNode):
             if not isinstance(val, CommentNode):
                 val = CommentNode(val)
             self.nodes.append(val)
```

Comment:
this `if` logic bothers me, but it works

Reply:
Yeah this seems jank. I'm trying to think about how to work around this.